### PR TITLE
Migrate to Log4j

### PR DESCRIPTION
**Pull request by subhramit**
Closes https://github.com/LambdaTest/lambdatest-gradle-plugin/issues/7

### CODE REVIEW

File: src/main/java/io/github/lambdatest/gradle/LambdaTestTask.java [hunk 1]
```diff
@@ -58,20 +62,20 @@ public void runLambdaTest() {
...
 
             if (testSuiteIdFuture != null) {
-                testSuiteId = testSuiteIdFuture.join(); 
-                System.out.println("Test suite uploaded successfully with ID: " + testSuiteId);
+                testSuiteId = testSuiteIdFuture.join();
+                logger.info("Test suite uploaded successfully with ID: {}", testSuiteId);
             }
         } catch (CompletionException e) {
-            System.err.println("Failed to execute tasks: " + e.getMessage());
+            logger.error("Failed to execute tasks: {}", e.getMessage());
```
Comment (harsha509) [inline src/main/java/io/github/lambdatest/gradle/LambdaTestTask.java:73]: Hi @subhramit,

Thank you for the PR.

You can just log full exception in here 
```suggestion
            logger.error("Failed to execute tasks: {}", e);
```

File: src/main/java/io/github/lambdatest/gradle/LambdaTestTask.java [hunk 2]
```diff
@@ -93,10 +97,10 @@ public void runLambdaTest() {
         try {
             testExecutor.executeTests(params);
         } catch (IOException e) {
-            System.err.println("Failed to execute tests: " + e.getMessage());
+            logger.error("Failed to execute tests: {}", e.getMessage());
```
Comment (harsha509) [inline src/main/java/io/github/lambdatest/gradle/LambdaTestTask.java:100]: same here may be!
```suggestion
            logger.error("Failed to execute tests: {}", e);
```

**Review by harsha509 — COMMENTED** [review]
HI @subhramit ,

Thank you for the PR!

I see that latest log4j is `2.24.1`, is there any specific reason that you've picked `2.14.1` which is 3 years old!

Thanks,
Sri

**Comment by subhramit** [comment]
> HI @subhramit ,
> 
> Thank you for the PR!
> 
> I see that latest log4j is `2.24.1`, is there any specific reason that you've picked `2.14.1` which is 3 years old!
> 
> Thanks,
> Sri

Hey, thanks for pointing that out!
You're right, I picked up the dependency lines from an old project template, and forgot to update it after being done with the changes.
I'll push a commit soon.

File: src/main/java/io/github/lambdatest/gradle/LambdaTestTask.java (hunk 2 above)
Comment (subhramit) [reply src/main/java/io/github/lambdatest/gradle/LambdaTestTask.java]: I committed your suggestions directly, left the rest as they are. Is there anywhere else I should log the exception fully?

Comment (harsha509) [reply src/main/java/io/github/lambdatest/gradle/LambdaTestTask.java]: no i think.!

**Review by harsha509 — APPROVED** [review]
Thank you @subhramit !

File: build.gradle [hunk 3]
```diff
@@ -15,6 +15,8 @@ dependencies {
     implementation 'com.google.code.gson:gson:2.8.8'
     implementation 'org.slf4j:slf4j-api:1.7.32'
     implementation 'ch.qos.logback:logback-classic:1.2.6'
+    implementation 'org.apache.logging.log4j:log4j-api:2.2.24.1'
+    implementation 'org.apache.logging.log4j:log4j-core:2.2.24.1'
```
Comment (navin772) [inline build.gradle:19]: A typo in version, please run the `gradle` tasks to verify dependencies after changing versions to verify. 

```suggestion
    implementation 'org.apache.logging.log4j:log4j-api:2.24.1'
    implementation 'org.apache.logging.log4j:log4j-core:2.24.1'
```

Comment (harsha509) [reply build.gradle:19]: Good catch, thank you @navin772 !

Comment (navin772) [reply build.gradle:19]: Command to verify dependencies - `./gradlew dependencies` 
Check for any FAILED dependency errors.

Comment (subhramit) [reply build.gradle]: > A typo in version, please run the `gradle` tasks to verify dependencies after changing versions to verify.

Hmm I assumed IntelliJ would warn if the dependency fetching fails. Resolved, build is successful now, thanks!

**Review by navin772 — APPROVED** [review]

**Comment by navin772** [comment]
@subhramit Thank you for your contribution, and congratulations on having your PR accepted for Hacktoberfest! :tada: Your effort helps improve the project, and we're excited to have contributors like you onboard. 
Keep up the great work, and happy coding! :blush::computer:

**Comment by subhramit** [comment]
> @subhramit Thank you for your contribution, and congratulations on having your PR accepted for Hacktoberfest! 🎉 Your effort helps improve the project, and we're excited to have contributors like you onboard. Keep up the great work, and happy coding! 😊💻

Thank you so much! Hope to learn more from you.

**Pull request merged**